### DISCUSSION TITLE
after trait is added to wrapper, but not loaded when reload

### DISCUSSION THREAD

**Discussion by xlnhaha** [GrapesJS/grapesjs#5251] [Q&A]
Hi,

The steps are as follows:

1, add trait "lang" to wrapper component
2, set lang="en" with addAttributes function
3, use getHtml(), and the result is like <body lang="en">, that is expected
4, then set lang=""
5, editor.setComponents(getHtml() of step 3), then check the lang trait value of wrapper, it is empty, this is not expected, i think this value should be "en"

please help, thank you very much.

**Comment by nguyenvoanhkhoa1** [comment]
I get the same problem
This is my code for adding traits, not in the init editor function:

  editor.Components.getWrapper().set("traits", [
    {
      type: "theme-select",
      label: "Theme color",
      name: "theme",
    },
  ]);
the wrapper setting panel has the trait at the first beginning (no data), when I select the option of the trait, the theme is applied. When I reset the page, the CSS of the theme existed and still applied, but the trait of the wrapper is not there anymore.

Any solution here?
Thanks a lot.

**Reply by artf** [reply]
Traits are not stored in the project data so, if you need to have persistent traits, you should extend the wrapper component (the code should be placed in a plugin) and not to be add dynamically
